### DISCUSSION TITLE
Build WASM

### DISCUSSION THREAD

**Discussion by NgVThangBz** [axmolengine/axmol#1350] [Q&A] [answered]
I'm very interested in WASM but I don't know much about them at the moment. Can you guide me on how to use it to create a website?

**Comment by halx99** [comment]
Very simple: just `axmol run -p wasm` , will build your project, then open local website in your browser

**Reply by NgVThangBz** [reply]
with:` axmol run -p wasm`
i got this error
![image_2023-09-22_15-41-50](https://github.com/axmolengine/axmol/assets/125545428/89ad81a6-4d3c-4e8b-bbf8-850c02ab209c)

**Reply by NgVThangBz** [reply]
As for mac, I get stuck here:
i tried again but it didn't work, and i also tried `axmol build -p wasm` and it kept crashing
![Screenshot 2023-09-22 at 14 58 52](https://github.com/axmolengine/axmol/assets/125545428/0c050724-2d80-448a-807b-c2eb62cb337e)

**Reply by halx99** [reply]
https://github.com/axmolengine/axmol/blob/dev/1k/build1k.ps1#L866

**Reply by NgVThangBz** [reply]
You mean you need to update the build1k file, I did that and
![image_2023-09-22_16-25-11](https://github.com/axmolengine/axmol/assets/125545428/5ca114a0-e274-4285-8906-67f171458792)

**Reply by halx99** [reply]
![image](https://github.com/axmolengine/axmol/assets/6977319/d3df7417-3072-40f8-871c-8cedf8379ba4)

please delete build_wasm then try again, i create a new project, and everything looks good.

**Reply by NgVThangBz** [reply]
im create new proj and run `axmol run -p wasm`  error still occurs

**Reply by halx99** [reply]
![image](https://github.com/axmolengine/axmol/assets/6977319/ea2330d8-47c7-4ee8-bcc0-33fe81c24ff6)

**Reply by halx99** [reply]
do you pull latest branch dev?

**Reply by NgVThangBz** [reply]
Yes, I just downloaded dev.zip and did it again,

**Reply by halx99** [reply]
i have no idea what happen with you, I can't reproduce this issue on my machine

**Reply by halx99** [reply, marked as answer]
emsdk require ninja setup, fixed by: https://github.com/axmolengine/axmol/commit/7b6b7fa29c199a32132763290c06f4a1829cd459

Note: emsdk also require a preinstalled python current need install manually

**Reply by NgVThangBz** [reply]
im using python 2.7 and 
`axmol run -p wasm`
![image](https://github.com/axmolengine/axmol/assets/125545428/ac41b9a0-f39d-4f2c-b3d8-42585e07e409)
![image](https://github.com/axmolengine/axmol/assets/125545428/0f4f6ff6-1cb5-4aae-ba9b-129a669e3ba6)

**Reply by halx99** [reply]
seems you directory case mismatch, please create new project try again

**Reply by halx99** [reply]
![image](https://github.com/axmolengine/axmol/assets/6977319/2f3fcd44-94f2-407b-a299-a5a06f0e8489)

or specificy build target name: `axmol build -p wasm -xb '--target,checkWasm'`

**Reply by NgVThangBz** [reply]
thank you, i do it

**Reply by halx99** [reply]
The commit will try read cmake target from  `CMakeCache.txt`, https://github.com/axmolengine/axmol/commit/6f079c07b464bedb260cf2bd632857c88e02f115

**Comment by halx99** [comment]
> thank you, i do it
